modules/porous_flow/include/materials/PorousFlowMaterialPermeabilityFromPorosity.h

```diff
@@ -0,0 +1,76 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+#define POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowMaterialPermeabilityFromPorosity;
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityFromPorosity>();
+
+/**
+ * Material designed to provide the permeability tensor
+ * which is a function of porosity
+ */
+class PorousFlowMaterialPermeabilityFromPorosity : public DerivativeMaterialInterface<Material>
+{
+public:
+  PorousFlowMaterialPermeabilityFromPorosity(const InputParameters & parameters);
+
+protected:
+  /// flags to indicate whether parameters have been set
+  bool _k0_set;
```

Comment:
you might not need this "*_set" parameters.  You could simply use parameters.isParamValid(...) in your "if" statements in your constructor.  If you still want to retain them, make them

const bool _k0_set;

The "const" indicates that you will never change them after construction, and the compiler will throw an error if you attempt it.